nexusutils/dataconverter/helpers.py

```diff
@@ -406,6 +409,10 @@ def validate_data_dict(template, data, nxdl_root: ET.Element):
                 index_of_at = nxdl_path.rindex("@")
                 nxdl_path = nxdl_path[0:index_of_at] + nxdl_path[index_of_at + 1:]
 
+            if nxdl_path in nxdl_path_to_type \
+               and isinstance(data[path], nxdl_path_to_type[nxdl_path]):
+                continue
+
```

Comment:
This doesn't work as this skips validation. We want to skip asking for the "elem" object below. You can store the elem object in this new dictionary you made. So wo won't need to call get_node_at_nxdl_path as this will be the same. But we still do want to verify that the entry is valid based on the little set of rules we check for.

Reply:
Using this if clause is the reason not to use get_node_at_nxdl_path(). You right we can store the element in the dict, I would argue that indeed it is. The nxdl_path is the conclusion/summary of the xml element. So, storing an unnecessary XML element will cost more space. @sanbrock also did objection on it. But after discussing we decided to keep it as it is now.